src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    if (this[0] !== null) {
```

Comment:
what abt `undefined` value? would be ieasier to move this check to your loop
![image](https://user-images.githubusercontent.com/37161315/173024873-c7514027-ba80-4aea-854c-35883d26ade4.png)